### PR TITLE
fix links in the documentation

### PR DESCRIPTION
**Pull request by keewis**
With the release of `sphinx` 3.3 we got a bit more control on the links in sections like `Returns` or `See Also`. Unfortunately, we still have the broken links from inherited docstrings, so unless we get upstream to change their docstrings we won't be able to fix that.

 - [x] Passes `isort . && black . && mypy . && flake8`

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -418,6 +418,7 @@
     "matplotlib": ("https://matplotlib.org", None),
     "dask": ("https://docs.dask.org/en/latest", None),
     "cftime": ("https://unidata.github.io/cftime", None),
+    "rasterio": ("https://rasterio.readthedocs.io/en/latest", None),
```
Comment (mathause) [inline doc/conf.py:421]: Should sparse be added? I don't know if there is already a crosslink but one is coming.

https://github.com/pydata/xarray/blob/2cde80bc100e26894f2328098f5b2139e4963e99/xarray/core/common.py#L1455

Comment (keewis) [reply doc/conf.py:421]: good point. I would do that in that PR, but if you think that is too confusing we can add this here.

Comment (mathause) [reply doc/conf.py:421]: :+1:  I asked to add it in the other PR, https://github.com/pydata/xarray/pull/4683#issuecomment-744064159

**Review by dcherian — APPROVED** [review]
LGTM

**Pull request merged**